### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by DaanWet**

### CODE REVIEW

File: src/main/java/commands/CommandHandler.java [hunk 1]
```diff
@@ -108,17 +113,19 @@ public void onCommandReceived(GuildMessageReceivedEvent e) throws Exception{
...
+                        break;
                     } else {
                         e.getMessage().delete().queueAfter(5, TimeUnit.SECONDS);
-                        channel.sendMessage("❌ You can't use that command here").queue(m -> m.delete().queueAfter(5, TimeUnit.SECONDS));
+                        channel.sendMessage(state.getError()).queue(m -> m.delete().queueAfter(5, TimeUnit.SECONDS));
                     }
-
                 } else {
                     e.getMessage().delete().queueAfter(5, TimeUnit.SECONDS);
                     channel.sendMessage("❌ You don't have permission to run this command").queue(m -> m.delete().queueAfter(5, TimeUnit.SECONDS));
```
Comment (Zelzahn) [inline src/main/java/commands/CommandHandler.java:131]: Can't `state.getError()` also be used here?

File: src/main/java/commands/casino/uno/Uno.java [hunk 2]
```diff
@@ -21,15 +24,14 @@ public Uno(GameHandler gameHandler) {
...
-
     @Override
     public void run(String[] args, GuildMessageReceivedEvent e) throws Exception {
         if (gameHandler.getUnoGame(e.getGuild().getIdLong()) != null) {
             e.getChannel().sendMessage("A game has already started").queue();
             return;
         }
         int bet = 0;
-        if (args.length == 1) {
+        if (args.length == 1) { // TODO: Check if betting is enabled
```
Comment (Zelzahn) [inline src/main/java/commands/casino/uno/Uno.java:34]: TODO

File: src/main/java/commands/settings/Settings.java [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                if (Utils.isInteger(value)) {
+                    dataHandler.setIntSetting(guildId, setting, subSetting, Utils.getInt(value), null);
+                } else {
+                    EmbedBuilder eb = new EmbedBuilder();
+                    eb.setTitle("Invalid input");
+                    eb.setColor(Color.RED);
+                    message.getChannel().sendMessage(eb.build()).queue();
+                }
+                break;
+            // TODO: CHANNEL_LONG AND ROLE_LONG
```
Comment (Zelzahn) [inline src/main/java/commands/settings/Settings.java:273]: TODO

File: src/main/java/commands/settings/Settings.java [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+                    } else {
+                        good = false;
+                        wrongValues.append(value).append(", ");
+                    }
+                }
+                if (!good) {
+                    wrongType = "a channel, role or member";
+                }
+                break;
+            // TODO: CHANNEL_LONG AND ROLE_LONG
```
Comment (Zelzahn) [inline src/main/java/commands/settings/Settings.java:339]: TODO

File: src/main/java/utils/DataHandler.java [hunk 5]
```diff
@@ -201,9 +244,9 @@ public int getCredits(long guildID, long userId) {
...
         try (Connection conn = DriverManager.getConnection(JDBC_URL, properties);
              PreparedStatement stm = conn.prepareStatement("INSERT IGNORE INTO Member(UserId, GuildId, LastDaily, LastWeekly) VALUES(?, ?, ?, ?);" +
-                     "UPDATE Member SET Credits = ? WHERE GuildId = ? AND UserId = ?;" +
-                     "INSERT IGNORE INTO UserRecord(UserId, GuildId, Name, Value) VALUES (?, ?, 'highest_credits', 0.0);" +
-                     "UPDATE UserRecord SET Value = GREATEST(Value, ?) WHERE UserId = ? AND GuildId = ? AND Name LIKE 'highest_credits'") //TODO: Use insert INto and ON Duplicate Keys
+                                                                   "UPDATE Member SET Credits = ? WHERE GuildId = ? AND UserId = ?;" +
+                                                                   "INSERT IGNORE INTO UserRecord(UserId, GuildId, Name, Value) VALUES (?, ?, 'highest_credits', 0.0);" +
+                                                                   "UPDATE UserRecord SET Value = GREATEST(Value, ?) WHERE UserId = ? AND GuildId = ? AND Name LIKE 'highest_credits'") //TODO: Use insert INto and ON Duplicate Keys
```
Comment (Zelzahn) [inline src/main/java/utils/DataHandler.java:249]: TODO

**Review by Zelzahn — APPROVED** [review]
Found a couple of very minor things and highlighted the remaining TODO's.

File: src/main/java/utils/DataHandler.java (hunk 5 above)
Comment (DaanWet) [reply src/main/java/utils/DataHandler.java:249]: Won't be worked on in this PR

**Pull request merged**